src/vdaf/mastic.rs

```diff
@@ -396,13 +396,55 @@ where
 /// parameters of Mastic used for encoding.
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct MasticPrepareState<F: FieldElement, const SEED_SIZE: usize> {
-    /// Includes output shares for eventual aggregation.
+    /// The counter and truncated weight for each candidate prefix.
     output_shares: MasticOutputShare<F>,
     /// If [`Szk`]` verification is being performed, we also store the relevant state for that operation.
     szk_query_state: SzkQueryState<SEED_SIZE>,
     verifier_len: Option<usize>,
 }
 
+impl<F: FieldElement, const SEED_SIZE: usize> Encode for MasticPrepareState<F, SEED_SIZE> {
+    fn encode(&self, bytes: &mut Vec<u8>) -> Result<(), CodecError> {
+        self.output_shares.encode(bytes)?;
+        if let Some(joint_rand_seed) = &self.szk_query_state {
+            joint_rand_seed.encode(bytes)?;
+        }
+        Ok(())
+    }
+
+    fn encoded_len(&self) -> Option<usize> {
+        Some(
+            self.output_shares.as_ref().len() * F::ENCODED_SIZE
+                + self.szk_query_state.as_ref().map_or(0, |_| SEED_SIZE),
+        )
+    }
+}
+
+impl<'a, T: Type, P: Xof<SEED_SIZE>, const SEED_SIZE: usize>
+    ParameterizedDecode<(&'a Mastic<T, P, SEED_SIZE>, &'a MasticAggregationParam)>
+    for MasticPrepareState<T::Field, SEED_SIZE>
+{
+    fn decode_with_param(
+        (mastic, agg_param): &(&Mastic<T, P, SEED_SIZE>, &MasticAggregationParam),
+        bytes: &mut Cursor<&[u8]>,
+    ) -> Result<Self, CodecError> {
+        let output_shares = MasticOutputShare::decode_with_param(&(*mastic, *agg_param), bytes)?;
```

Comment:
This may be a lot of data to store, since the length of `output_shares` is linear in the number of candidate prefixes. In most cases, it would be more compact to store the VIDPF seed and correction words along with the candidate prefixes.  However, re-inflating would cost CPU.